Fastfile

```diff
@@ -4,77 +4,178 @@ default_platform(:ios)
+      env_variables[:core][:workspace],
+      env_variables[:core][:app_name],
+      env_variables[:core][:slack_webhook_url],
+      env_variables[:s3][:bucket],
+      env_variables[:s3][:access_key],
+      env_variables[:s3][:secret_access_key],
+      env_variables[:s3][:region],
+      env_variables[:match][:key_id],
+      env_variables[:match][:issuer_id],
+      env_variables[:match][:key_content],
+      env_variables[:match][:in_house],
+      env_variables[:match][:match_password]
+    ],
+
+    store: [
+      env_variables[:core][:app_id],
+      env_variables[:core][:scheme],
+      env_variables[:core][:ipa_name],
+      env_variables[:core][:info_plist],
+      env_variables[:core][:app_store_build_config],
+      env_variables[:core][:itc_name]
+    ],
+
+    prod: [
+      env_variables[:core][:app_id],
+      env_variables[:core][:prod_scheme],
+      env_variables[:core][:prod_ipa_name],
+      env_variables[:core][:prod_tryouts_app_id],
+      env_variables[:core][:prod_tryouts_api_token],
+      env_variables[:core][:prod_info_plist],
+      env_variables[:core][:adhoc_build_config]
+    ],
+
+    preprod: [
+      env_variables[:core][:preprod_app_id],
+      env_variables[:core][:preprod_scheme],
+      env_variables[:core][:preprod_ipa_name],
+      env_variables[:core][:preprod_tryouts_app_id],
+      env_variables[:core][:preprod_tryouts_api_token],
+      env_variables[:core][:preprod_info_plist],
+      env_variables[:core][:adhoc_build_config]
+    ],
+
+    staging: [
+      env_variables[:core][:staging_app_id],
+      env_variables[:core][:staging_scheme],
+      env_variables[:core][:staging_ipa_name],
+      env_variables[:core][:staging_tryouts_app_id],
+      env_variables[:core][:staging_tryouts_api_token],
+      env_variables[:core][:staging_info_plist],
+      env_variables[:core][:adhoc_build_config]
+    ]
+  }
 
   before_all do |lane, options|
-    shared_env_variables = [
-      "FASTLANE_USER",
-      "TEAM_ID",
-      "KEY_ID",
-      "ISSUER_ID",
-      "KEY_CONTENT",
-      "IN_HOUSE",
-      "WORKSPACE",
-      "SLACK_WEBHOOK_URL",
-      "MATCH_PASSWORD",
-      "S3_BUCKET",
-      "S3_ACCESS_KEY",
-      "S3_SECRET_ACCESS_KEY",
-      "S3_REGION",
-      "APP_NAME"
-    ]
+    check_env_variables(variables: required_env_variables[:shared])
+
+    if env_value(key: env_variables[:core][:app_id]) != nil
+      check_env_variables(variables: required_env_variables[:store])
+      check_env_variables(variables: required_env_variables[:prod])
+    end
+
+    if env_value(key: env_variables[:core][:preprod_app_id]) != nil
+      check_env_variables(variables: required_env_variables[:preprod])
+    end
+
+    if env_value(key: env_variables[:core][:staging_app_id]) != nil
+      check_env_variables(variables: required_env_variables[:staging])
+    end
+  end
+
+  lane :check_env_variables do |options|
+    ensure_env_vars(
+      env_vars: options[:variables]
+    )
```

Comment:
I think you can remove this method because its signature is the same with the inner one.